### ISSUE TITLE
Rule idea: warn when checkModules()/verify() is used without androidContext() mock

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#57]
## Problem

`checkModules()` / `verify()` fails silently when modules use `androidContext()` but no Android context is provided in test setup.

## Detection Pattern

```kotlin
@Test
fun `verify modules`() {
    val koin = koinApplication {
        modules(appModule)  // ❌ Fails if appModule uses androidContext()
    }
    koin.checkModules()
}
```

## Related Koin Issues
- https://github.com/InsertKoinIO/koin/issues/553

## Implementation Notes
Low priority. May be better addressed as documentation than a static analysis rule.
